[docs][material-ui] Improve `theme.applyStyles()` docs (#44658)
mui/material-ui: docs/data/material/customization/dark-mode/dark-mode.md

</Button>;
```

> ### Codemod

Author: This section wasn't removed, it was just moved below

</Button>;
```

### API

`theme.applyStyles(mode, styles) => CSSObject`

Apply styles for a specific mode.

#### Arguments

- `mode` (`'light' | 'dark'`) - The mode for which the styles should be applied.

> - `styles` (`CSSObject`) - An object which contains the styles to be applied for the specified mode.

Reviewer: ```suggestion
- `styles` (`CSSObject`) - An object that contains the styles to be applied for the specified mode.
```

#### Returns

> A `CSSObject` which key is the mode selector and value is the styles object: `{ [modeSelector]: styles }`.

Reviewer 2: This is true if the CSS theme variables feature is enable. However, the default behavior is still using `theme.palette.mode` to override the styles.

```js
{
  color: 'red',
  …theme.applyStyles('dark', {
    color: 'blue',
  }),
}

// result when theme.palette.mode === 'dark' is
{
  color: 'blue',
}
```

> A `CSSObject` which key is the mode selector and value is the styles object: `{ [modeSelector]: styles }`.

Reviewer: ```suggestion
A `CSSObject` within which the key is the mode selector and the value is the styles object: `{ [modeSelector]: styles }`.
```

Author: I removed this section in the latest commit

By default, the `modeSelector` is `'@media (prefers-color-scheme: light)'` for light mode and `'@media (prefers-color-scheme: dark)'` for dark mode.

### Specificity

> Styles applied with `theme.applyStyles()` have higher specificity than the styles defined outside of it.

Reviewer: ```suggestion
Styles applied with `theme.applyStyles()` have higher specificity than those defined outside of it.
```

> This means that to override styles applied with it (via the `styled` function or `sx` prop), you need to use `theme.applyStyles()` as well:

Reviewer: Nitpick - the original wording is fine as is, it's just kind of tough to get the phrasing to not sound awkward. This was my best attempt:

```suggestion
So if you need to override styles from this function via the `styled` function or `sx` prop, you must include it as shown below:
```

Author: Reworded

```jsx
const BaseButton = styled('button')(({ theme }) =>
  theme.applyStyles('dark', {
    backgroundColor: 'white',
  }),
);

const AliceblueButton = styled(BaseButton)({
  backgroundColor: 'aliceblue', // In dark mode, backgroundColor will be white as theme.applyStyles() has higher specificity
});

const PinkButton = styled(BaseButton)(({ theme }) =>
  theme.applyStyles('dark', {
    backgroundColor: 'pink', // In dark mode, backgroundColor will be pink
  }),
);
```

### Overriding applyStyles

> If you need to override `theme.applyStyles()`, you can do so by providing a custom function.

Reviewer: this feels strange given what comes right before it—we present a way to override this function in the Specificity section, then suggest a different way to do it here. Or I guess this is describing a more specific use case—overriding styles vs overriding the way the function itself works?

Author: Moved the last line which clarifies this to the top

> A case in which you might need to do it is to return a string instead of an object so it can be used inside template literals:

Reviewer: ```suggestion
For instance, if you need the function to return a string instead of an object so it can be used inside template literals:
```

```js
const theme = createTheme({
  cssVariables: {
    colorSchemeSelector: '.mode-%s',
  },
  colorSchemes: {
    dark: {},
    light: {},
  },
  applyStyles: function (key: string, styles: any) {
    // return a string instead of an object
    return `*:where(.mode-${key}) & {${styles}}`;
  },
});

const StyledButton = styled('button')`
  ${theme.applyStyles(
    'dark', `
      background: white;
    `
  )}
`;
```

> When overriding `theme.applyStyles()`, you have complete control over its arguments and return value.

Reviewer: ah ok this sentence makes things more clear. Maybe it should go up at the top of this section.

#### Arguments

- `mode` (`'light' | 'dark'`) - The mode for which the styles should be applied.
- `styles` (`CSSObject`) - An object that contains the styles to be applied for the specified mode.

#### Overriding applyStyles

> You can override `theme.applyStyles()` with a custom function to have complete control over it.

Reviewer: Nitpick: to be more specific about what's meant by "have complete control over it."

```suggestion
You can override `theme.applyStyles()` with a custom function to gain complete control over the values it returns.
```
